src/app/components/primitives/sidebar.tsx

```diff
@@ -0,0 +1,204 @@
+      icon: pathname.toLowerCase().startsWith("/dashboard") ? (
+        <OverviewSolidIcon className={`w-5 h-5`} />
+      ) : (
+        <OverviewOutlineIcon className={`w-5 h-5`} />
+      ),
+      label: "Overview",
+    },
+    {
+      href: "/challenges",
+      icon: pathname.toLowerCase().startsWith("/challenges") ? (
+        <PuzzlePieceSolidIcon className={`w-5 h-5`} />
+      ) : (
+        <PuzzlePieceOutlineIcon className={`w-5 h-5`} />
+      ),
+      label: "Challenges",
+    },
+    {
+      href: "/leaderboard",
+      icon: pathname.toLowerCase().startsWith("/leaderboard") ? (
+        <TrophySolidIcon className={`w-5 h-5`} />
+      ) : (
+        <TrophyOutlineIcon className={`w-5 h-5`} />
+      ),
+      label: "Leaderboard",
+    },
+    {
+      href: "/notifications",
+      icon: pathname.toLowerCase().startsWith("/notifications") ? (
+        <BellSolidIcon className="w-5 h-5" />
+      ) : (
+        <BellOutlineIcon className="w-5 h-5" />
+      ),
+      label: "Notifications",
+    },
+    {
+      href: "/profile",
+      icon: pathname.toLowerCase().startsWith("/profile") ? (
+        <UserSolidIcon className="w-5 h-5" />
+      ) : (
+        <UserOutlineIcon className="w-5 h-5" />
+      ),
+      label: "Profile",
+    },
+  ];
+
+  return (
+    <>
+      {isHamburgerOpen && (
+        <div
+          className="lg:hidden fixed inset-0 bg-black bg-opacity-50 z-20"
+          onClick={() => setIsHamburgerOpen(false)}
+        />
+      )}
+      <aside
+        className={`fixed lg:relative ${
+          isHamburgerOpen ? "flex" : "hidden"
+        } lg:flex h-screen flex-col border-r bg-white z-30 transition-all duration-300 ${
+          !isOpen
+            ? "w-0 lg:w-[104px] -translate-x-full lg:translate-x-0"
+            : "w-[266px]"
+        }`}
+      >
+        <div className="relative flex items-center h-24 justify-between border-b border-grey-button-border">
+          <button
+            onClick={() => setIsOpen(!isOpen)}
+            className="hidden lg:block absolute -right-5 p-3 rounded-full border border-grey-button-border bg-white"
+          >
+            <DoubleChevronLeft
+              className={`w-[14px] h-[14px] ${isOpen ? "" : "rotate-180"}`}
+            />
+          </button>
+          <Link
+            href="/dashboard"
+            className={`flex items-center m-0 p-0 ${isOpen ? "ml-3" : "ml-7"}`}
+            onClick={() => {
+              setIsHamburgerOpen(false);
+            }}
+          >
+            <JedeIcon
+              withText={isOpen}
+              className={`${isOpen ? "w-32" : "w-10 h-10"}`}
+            />
+          </Link>
+        </div>
+
+        <nav className="flex-1 items-center justify-center space-y-8">
+          {navItems.map((item) => (
+            <NavItem
+              key={item.href}
+              href={item.href}
+              icon={item.icon}
+              label={isOpen || window.innerWidth < 1024 ? item.label : ""}
```

Comment:
We should probably move this `window.innerWidth` check out to a useEffect as you have done in the `ProfileDropdown` component.  The reason is to make it responsive to screen size changes after rendering.

Reply:
Its actually not needed so I'll remove it